Frameworks/Foundation/NSObject.mm

```diff
@@ -304,16 +313,29 @@ - (IMP)methodForSelector:(SEL)selector {
 /**
  @Status Interoperable
 */
-- (NSMethodSignature*)methodSignatureForSelector:(SEL)selector {
-    return [[self class] instanceMethodSignatureForSelector:selector];
++ (NSMethodSignature*)methodSignatureForSelector:(SEL)selector {
```

Comment:
This has to exist so that we don't need to complicate instanceMethodForSelector for metaclassses. If we used the same codepath, we'd lose the ability to call `resolveClassMethod` on the appropriate target (the class)